test/compress/issue-203.js

```diff
@@ -26,9 +26,7 @@ compress_new_function_with_destruct: {
     }
     expect: {
         Function("a", "[b]", "return a");
-        Function("a", "{bb}", "return a");
-        Function("[[a]]", "[{bb}]", 'return a');
+        Function("a", "{bb:b}", "return a");
+        Function("[[a]]", "[{bb:b}]", 'return a');
```

Comment:
As mentioned in https://github.com/mishoo/UglifyJS2/pull/1464#issuecomment-277434245, this use case is rare and unusual, so I'm not sure if we want to spend too much effort on this unless it affects correctness of output.